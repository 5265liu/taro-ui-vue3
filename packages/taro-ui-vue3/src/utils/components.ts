
export const View = 'taro-view-core'
export const Icon = 'taro-icon-core'
export const Progress = 'taro-progress-core'
export const Text = 'taro-text-core'
export const Button = 'taro-button-core'
export const Checkbox = 'taro-checkbox-core'
export const CheckboxGroup = 'taro-checkbox-group-core'
export const Form = 'taro-form-core'
export const Input = 'taro-input-core'
export const Label = 'taro-label-core'
export const OpenData = 'taro-open-data-core'
export const Picker = 'taro-picker-core'
export const PickerView = 'taro-picker-view-core'
export const PickerViewColumn = 'taro-picker-view-column-core'
export const Radio = 'taro-radio-core'
export const RadioGroup = 'taro-radio-group-core'
export const Slider = 'taro-slider-core'
export const Switch = 'taro-switch-core'
export const Textarea = 'taro-textarea-core'
export const ScrollView = 'taro-scroll-view-core'
export const Swiper = 'taro-swiper-core'
export const SwiperItem = 'taro-swiper-item-core'
export const Image = 'taro-image-core'