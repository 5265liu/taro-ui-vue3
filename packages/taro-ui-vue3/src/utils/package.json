{
  "name": "@taro-ui-vue3/utils",
  "version": "0.0.0",
  "license": "MIT",
  "main": "common.js",
  "peerDependencies": {
    "vue": "^3.0.0",
    "@tarojs/taro": "^3.0.0"
  }
}