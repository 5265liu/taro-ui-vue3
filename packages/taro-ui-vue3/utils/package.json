{
  "name": "@taro-ui-vue3/utils",
  "version": "0.0.0",
  "license": "MIT",
  "main": "common.js",
  "peerDependencies": {
    "@tarojs/taro": "^3.0.0"
  }
}