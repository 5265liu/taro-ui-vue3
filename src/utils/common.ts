import Taro, { SelectorQuery } from '@tarojs/taro'

const { getEnv, ENV_TYPE } = Taro
const ENV = Taro.getEnv()

type ENVS = {
    isWEAPP: boolean,
    isALIPAY: boolean,
    isWEB: boolean,
}

const getEnvs = (): ENVS => {
    const env = getEnv()

    return {
        isWEAPP: env === ENV_TYPE.WEAPP,
        isALIPAY: env === ENV_TYPE.ALIPAY,
        isWEB: env === ENV_TYPE.WEB,
    }    
}

function pxTransform(size: number): string {
    if (!size) return ''

    const designWidth = 750
    const deviceRatio = {
        640: 2.34 / 2,
        750: 1,
        828: 1.81 / 2,
    }

    return `${size / deviceRatio[designWidth]}rpx`
}

function delay(delayTime = 500): Promise<null> {
    return new Promise((resolve) => {
        if ([Taro.ENV_TYPE.WEB, Taro.ENV_TYPE.SWAN].includes(ENV)) {
            setTimeout(() => {
                resolve()
            }, delayTime)
            return
        }
        resolve()
    })
}

function delayQuerySelector(_, selectorStr: string, delayTime = 500): Promise<[]> {
    false && console.log(_)
    const selector: SelectorQuery = Taro.createSelectorQuery()

    return new Promise((resolve) => {
        delay(delayTime).then(() => {
            selector
                .select(selectorStr)
                .boundingClientRect()
                .exec((res: []) => {
                    resolve(res)
                })
        })
    })
}


export { 
    pxTransform,
    getEnvs,
    delayQuerySelector,
}