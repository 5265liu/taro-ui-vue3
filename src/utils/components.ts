import { getEnvs } from './common'
import {
    View as TaroView,
    Icon as TaroIcon,
    Progress as TaroProgress,
    Text as TaroText,
    Button as TaroButton,
    Checkbox as TaroCheckBox,
    CheckboxGroup as TaroCheckBoxGroup,
    Form as TaroForm,
    Input as TaroInput,
    Label as TaroLabel,
    OpenData as TaroOpenData,
    Picker as TaroPicker,
    PickerView as TaroPickerView,
    PickerViewColumn as TaroPickerViewColumn,
    Radio as TaroRadio,
    RadioGroup as TaroRadioGroup,
    Slider as TaroSlider,
    Switch as TaroSwitch,
    Textarea as TaroTextarea,
    ScrollView as TaroScrollView,
    Swiper as TaroSwiper,
    SwiperItem as TaroSwiperItem,
    Image as TaroImage
} from "@tarojs/components";

const isWeb = getEnvs().isWEB

export const View = isWeb ? 'view' : TaroView
export const Icon = isWeb ? 'icon' : TaroIcon
export const Progress = isWeb ? 'progress' : TaroProgress
export const Text = isWeb ? 'text' : TaroText
export const Button = isWeb ? 'button' : TaroButton
export const Checkbox = isWeb ? 'checkbox' : TaroCheckBox
export const CheckboxGroup = isWeb ? 'checkbox-group' : TaroCheckBoxGroup
export const Form = isWeb ? 'form' : TaroForm
export const Input = isWeb ? 'input' : TaroInput
export const Label = isWeb ? 'label' : TaroLabel
export const OpenData = isWeb ? 'open-data' : TaroOpenData
export const Picker = isWeb ? 'picker' : TaroPicker
export const PickerView = isWeb ? 'picker-view' : TaroPickerView
export const PickerViewColumn = isWeb ? 'picker-view-column' : TaroPickerViewColumn
export const Radio = isWeb ? 'radio' : TaroRadio
export const RadioGroup = isWeb ? 'radio-group' : TaroRadioGroup 
export const Slider = isWeb ? 'slider' : TaroSlider
export const Switch = isWeb ? 'switch' : TaroSwitch
export const Textarea = isWeb ? 'textarea' : TaroTextarea
export const ScrollView = isWeb ? 'scroll-view' : TaroScrollView
export const Swiper = isWeb ? 'swiper' : TaroSwiper
export const SwiperItem = isWeb ? 'swiper-item' : TaroSwiperItem
export const Image = isWeb ? 'img' : TaroImage
