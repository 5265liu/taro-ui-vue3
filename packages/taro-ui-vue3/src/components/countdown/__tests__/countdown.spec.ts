import { mount } from '@vue/test-utils'
import AtCountdown from '../index.vue'
import { sleep } from '@/utils/testHelper'

const factory = async (values = {}, slots = { default: [] }) => {
  const component = mount(AtCountdown, {
    slots,
    props: { ...values },
  })
  await sleep(0)
  return component
}

describe('AtCountdown Snap', () => {
  it('render initial AtCountdown', async () => {
    const wrapper = await factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props className', async () => {
    const wrapper = await factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props customStyle', async () => {
    const wrapper = await factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props isCard', async () => {
    const wrapper = await factory({ isCard: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props format', async () => {
    const wrapper = await factory({ format: { hours: ':', minutes: ':', seconds: '' } })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props day', async () => {
    const wrapper = await factory({ day: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props hours', async () => {
    const wrapper = await factory({ hours: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props minutes', async () => {
    const wrapper = await factory({ minutes: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props seconds', async () => {
    const wrapper = await factory({ seconds: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props isShowDay', async () => {
    const wrapper = await factory({ isShowDay: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- hours > 24 and show day', async () => {
    const wrapper = await factory({
      isShowDay: true,
      hours: 25,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- hours > 24 and not show day', async () => {
    const wrapper = await factory({
      isShowDay: false,
      hours: 25,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
