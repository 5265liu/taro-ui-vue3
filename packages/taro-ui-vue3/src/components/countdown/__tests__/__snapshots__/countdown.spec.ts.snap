// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtCountdown Snap render AtCountdown -- hours > 24 and not show day 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        25
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- hours > 24 and show day 1`] = `
<view
  class="at-countdown"
>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      天
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props className 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props customStyle 1`] = `
<view
  class="at-countdown"
  customstyle="color:red;"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props day 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        24
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props format 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      :
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      :
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    />
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props hours 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props isCard 1`] = `
<view
  class="at-countdown--card at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props isShowDay 1`] = `
<view
  class="at-countdown"
>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      天
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props minutes 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props seconds 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render initial AtCountdown 1`] = `
<view
  class="at-countdown"
>
  <!--v-if-->
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <!-- countdown time-box -->
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <!-- separator -->
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;
